api/src/main/resources/liquibase.xml

```diff
@@ -0,0 +1,320 @@
+			</column>
+			<column name="date_changed" type="datetime" defaultValue="NULL">
+				<constraints nullable="true"/>
+			</column>
+			<column name="retired" type="tinyint(1)" defaultValue="0">
+				<constraints nullable="false"/>
+			</column>
+			<column name="retired_by" type="int(11)" defaultValue="NULL">
+				<constraints nullable="true" unique="true" foreignKeyName="retired_by for idgen_identifier_source" references="users(user_id)"/>
+			</column>
+			<column name="date_retired" type="datetime" defaultValue="NULL">
+				<constraints nullable="true"/>
+			</column>
+			<column name="retire_reason" type="varchar(255)" defaultValue="NULL">
+				<constraints nullable="true"/>
+			</column>
+		</createTable>
+	</changeSet>
+
+	<changeSet id="a7a39c84-cb48-427e-9141-ee851c283132" author="Partners In Health">
+		<preConditions onFail="MARK_RAN">
+			<not><tableExists tableName="idgen_seq_id_gen" /></not>
+		</preConditions>
+		<createTable tableName="idgen_seq_id_gen">
+			<column name="id" type="int(11)">
+				<constraints nullable="false" primaryKey="true" foreignKeyName="id for idgen_seq_id_gen" references="idgen_identifier_source(id)"/>
+			</column>
+			<column name="next_sequence_value" type="int(11)" defaultValue="-1">
+				<constraints nullable="false"/>
+			</column>
+			<column name="base_character_set" type="varchar(255)">
+				<constraints nullable="false"/>
+			</column>
+			<column name="first_identifier_base" type="varchar(50)">
+				<constraints nullable="false"/>
+			</column>
+			<column name="prefix" type="varchar(20)">
+				<constraints nullable="false"/>
+			</column>
+			<column name="suffix" type="varchar(20)">
+				<constraints nullable="false"/>
+			</column>
+			<column name="length" type="int(11)">
+				<constraints nullable="false"/>
+			</column>
+		</createTable>
+	</changeSet>
+
+	<changeSet id="a92f91c0-162b-42d9-a33b-c1412dc2948d" author="Partners In Health">
+		<preConditions onFail="MARK_RAN">
+			<not><tableExists tableName="idgen_remote_source" /></not>
+		</preConditions>
+		<createTable tableName="idgen_remote_source">
+			<column name="id" type="int(11)">
+				<constraints nullable="false" primaryKey="true" foreignKeyName="id for idgen_remote_source" references="idgen_identifier_source(id)"/>
+			</column>
+			<column name="url" type="varchar(255)">
+				<constraints nullable="false"/>
+			</column>
+		</createTable>
+	</changeSet>
+
+	<changeSet id="b50c8351-3c6c-40eb-be40-bbdaed78c5a0" author="Partners In Health">
+		<preConditions onFail="MARK_RAN">
+			<not><tableExists tableName="idgen_id_pool" /></not>
+		</preConditions>
+		<createTable tableName="idgen_id_pool">
+			<column name="id" type="int(11)">
+				<constraints nullable="false" primaryKey="true" foreignKeyName="id for idgen_id_pool" references="idgen_identifier_source(id)"/>
+			</column>
+			<column name="source" type="int(11)">
```

Comment:
Is this unique?

Reply:
It thought it should be because the column has a foreignKey constraint

Reply:
Are you sure foreign key constraint fields should be unique in the referencing table?